pkg/volume/azure_file/azure_file.go

```diff
@@ -139,6 +142,45 @@ func (plugin *azureFilePlugin) newUnmounterInternal(volName string, podUID types
 	}}, nil
 }
 
+func (plugin *azureFilePlugin) RequiresFSResize() bool {
+	return false
+}
+
+func (plugin *azureFilePlugin) ExpandVolumeDevice(
+	spec *volume.Spec,
+	newSize resource.Quantity,
+	oldSize resource.Quantity) (resource.Quantity, error) {
+
+	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.AzureFile == nil {
+		return oldSize, fmt.Errorf("invalid PV spec")
+	}
+	shareName := spec.PersistentVolume.Spec.AzureFile.ShareName
+	azure, err := getAzureCloudProvider(plugin.host.GetCloudProvider())
+	if err != nil {
+		glog.V(2).Infof("failed to get azure provider")
```

Comment:
Doesn't caller log the returned error anyways? so - rather than logging the similar error from two places, it may be better to augment the original and return a modified value via `fmt.Errorf` ?

Reply:
good catch, actually it's not necessary, I have removed those info logs